### PR TITLE
Add securityContext field to all the Weaviate StatefulSet and all the modules Deployments 

### PR DESCRIPTION
**Pull request by StefanBogdan**

### CODE REVIEW

File: weaviate/templates/weaviateStatefulset.yaml [hunk 1]
```diff
@@ -43,7 +45,7 @@ spec:
           privileged: true
         image: "{{ .image.registry }}/{{ .image.repo }}:{{ .image.tag }}"
         imagePullPolicy: "{{ .image.pullPolicy }}"
-        command: ["sysctl", "vm.max_map_count={{ .sysctlVmMaxMapCount }}", "vm.overcommit_memory=1" ]
+        command: ["sysctl", "-w", "vm.max_map_count={{ .sysctlVmMaxMapCount }}", "vm.overcommit_memory=1" ]
```
Comment (StefanBogdan) [inline weaviate/templates/weaviateStatefulset.yaml:48]: I thought that this was already fixed, no Idea what happened to the fix.

File: weaviate/values.yaml [hunk 2]
```diff
@@ -702,6 +715,17 @@ modules:
...
+
+    # security Context for the Contextionary Pods. The configurations are the same as setting them
+    # as described here: https://kubernetes.io/docs/tasks/configure-pod-container/security-context/
+    securityContext: {}
+
+    # It is possible to add a ServiceAccount to this module's Pods, it can be
+    # used in cases where the module is in a private registry and you want to
+    # give access to the registry only to this pod.
+    # NOTE: if not set the root `serviceAccountName` config will be used.
+    serviceAccountName:
```
Comment (StefanBogdan) [inline weaviate/values.yaml:727]: This seems like something that I forgot to add in the previous PR.

File: weaviate/templates/weaviateStatefulset.yaml (hunk 1 above)
Comment (StefanBogdan) [reply weaviate/templates/weaviateStatefulset.yaml:48]:
```
COMPATIBILITY
     The -w option has been deprecated and is silently ignored.
```
Hmmm

**Review by antas-marcin — APPROVED** [review]

File: weaviate/templates/weaviateStatefulset.yaml (hunk 1 above)
Comment (mikewyer) [reply weaviate/templates/weaviateStatefulset.yaml:48]: FWIW `-w` is deprecated on MacOS, and I checked the sources for ubuntu and Alpine linux and they both have the logic `if (WriteMode || strchr(*argv, '='))` so the `-w` should be redundant.

Comment (rbtz-openai) [reply weaviate/templates/weaviateStatefulset.yaml:48]: JFYI: it was changed here: https://github.com/weaviate/weaviate-helm/pull/149

It should not work with `-w` in theory and it indeed breaks init container in our deployments.

Comment (StefanBogdan) [reply weaviate/templates/weaviateStatefulset.yaml:48]: The docs say that it is deprecated and is ignored. @rbtz-openai so why does it break for you?

I can release a new patch without the `-w`.

**Pull request merged**